### ISSUE TITLE
AutoGraph "could not transform" warning when code contains a multi-line string with backslashes

### ISSUE THREAD

**Issue by dmtlvn** [tensorflow/tensorflow#40589]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution: Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.3
- CUDA/cuDNN version: 10.1/7
- GPU model and memory: Nvidia GeForce RTX 2080Ti

**Describe the current behavior**
AutoGraph warning appears if a custom keras layer's `call` method code contains a multiline string joined by the backslash. If a multiline string is joined using brackets, however, no warning appears. Does not seem to influence calculations in any way, but a fun bug to encounter

```
WARNING:tensorflow:AutoGraph could not transform <bound method MyLayer.call of <__main__.MyLayer object at 0x0000XXXXXXXXXXXX>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: expected exactly one node node, found [<gast.gast.FunctionDef object at 0x0000XXXXXXXXXXXX>, <gast.gast.Return object at 0x0000XXXXXXXXXXXX>]
WARNING: AutoGraph could not transform <bound method MyLayer.call of <__main__.MyLayer object at 0x0000XXXXXXXXXXXX>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: expected exactly one node node, found [<gast.gast.FunctionDef object at 0x0000XXXXXXXXXXXX>, <gast.gast.Return object at 0x0000XXXXXXXXXXXX>]
```

**Describe the expected behavior**
No warnings appears

**Standalone code to reproduce the issue**
```
import tensorflow as tf
tf.autograph.set_verbosity(10, alsologtostdout = True)

from tensorflow.keras.layers import Layer, Input

class SlashPhobic(Layer):

    def call(self, inputs):
        s = "foo" \
            "bar"
        print(s)
        return inputs

x = Input(shape = (1,))
y = SlashPhobic()(x)
```

Autograph log:

```
Converted call: <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>>
    args: (<tf.Tensor 'input_1:0' shape=(None, 1) dtype=float32>,)
    kwargs: {}

Not whitelisted: <method-wrapper '__call__' of method object at 0x0000XXXXXXXXXXXX>: default rule
Not whitelisted: <class '__main__.SlashPhobic'>: default rule
Not whitelisted: <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>>: default rule
Entity <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>> is not cached for key <code object call at 0x0000XXXXXXXXXXXX, file ".\phobic.py", line 8> subkey (<tensorfl
ow.python.autograph.core.converter.ConversionOptions object at 0x0000XXXXXXXXXXXX>, frozenset())
Converting <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>>
Error transforming entity <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>>
Traceback (most recent call last):
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\impl\api.py", line 526, in converted_call
    converted_f = conversion.convert(target_entity, program_ctx)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\impl\conversion.py", line 326, in convert
    free_nonglobal_var_names)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\impl\conversion.py", line 240, in _convert_with_cache
    entity, program_ctx)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\impl\conversion.py", line 475, in convert_entity_to_ast
    nodes, name, entity_info = convert_func_to_ast(o, program_ctx)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\impl\conversion.py", line 634, in convert_func_to_ast
    node, source = parser.parse_entity(f, future_features=future_features)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\pyct\parser.py", line 207, in parse_entity
    return _attempt_to_parse_normal_source(source, future_features)
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\pyct\parser.py", line 111, in _attempt_to_parse_normal_source
    return parse_str(source, preamble_len=len(future_features)), source
  File "D:\Miniconda3\lib\site-packages\tensorflow_core\python\autograph\pyct\parser.py", line 230, in parse_str
    raise ValueError('expected exactly one node node, found {}'.format(nodes))
ValueError: expected exactly one node node, found [<gast.gast.FunctionDef object at 0x0000XXXXXXXXXXXX>, <gast.gast.Return object at 0x0000XXXXXXXXXXXX>]
WARNING:tensorflow:AutoGraph could not transform <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: expected exactly one node node, found [<gast.gast.FunctionDef object at 0x0000XXXXXXXXXXXX>, <gast.gast.Return object at 0x0000XXXXXXXXXXXX>]
WARNING: AutoGraph could not transform <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x0000XXXXXXXXXXXX>> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: expected exactly one node node, found [<gast.gast.FunctionDef object at 0x0000XXXXXXXXXXXX>, <gast.gast.Return object at 0x0000XXXXXXXXXXXX>]
```

**Comment by Saduf2019** [comment]
@dmtlvn 
I ran the code shared on tf 2.1,2.2 and tf nightly and see this error, can you please confirm if [this is the issue](https://colab.research.google.com/gist/Saduf2019/7c6cbdb1a4b0bf8d749b4590c0f462ac/untitled238.ipynb) faced.

**Comment by dmtlvn** [comment]
@Saduf2019 
The code in the colab seems to work fine, so I am not sure what issue you are talking about. Looks like colab has some preprocessor, which changes the code to compile and removes the backslash:
```
INFO:tensorflow:Source code of <bound method SlashPhobic.call of <__main__.SlashPhobic object at 0x7f074ff23b38>>:

def call(self, inputs):
    s = "foo"             "bar"
    print(s)
    return inputs
```

However, I am able to reproduce this warning in a separate environment by running a script with the code above, as well as running a jupyter notebook.

**Comment by mdanatg** [comment]
This looks the same as #35765. It should be fixed in TF 2.2.
